webapp/packages/core-di/src/App.ts

```diff
@@ -69,17 +77,16 @@ export class App {
   }
 
   // first phase register all dependencies
-  private registerServices(): void {
-    this.getServiceCollection().addServiceByClass(App, this);
-
-    for (const service of this.getServices()) {
-      // console.log('provider', provider.name);
+  private async registerServices(preload?: boolean): Promise<void> {
+    for (const serviceLoader of this.getServices(preload)) {
+      const service = await serviceLoader();
       this.diWrapper.collection.addServiceByClass(service);
     }
```

Comment:
now we register services one by one which reduces waiting time
can we painfulessly register it in parallel mode like this?

```suggestion
    this.getServices(preload).forEach(serviceLoader => {
      serviceLoader().then(service => {
        this.diWrapper.collection.addServiceByClass(service);
      });
    });
```

Reply:
agree